python/lsst/pipe/base/pipeline_graph/__main__.py

```diff
@@ -0,0 +1,583 @@
+# This file is part of pipe_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ()
+
+import argparse
+import dataclasses
+import sys
+import textwrap
+from collections.abc import Sequence
+from contextlib import ExitStack
+
+from lsst.daf.butler import Butler
+from lsst.resources import ResourcePath
+
+from ..pipeline import Pipeline
+from ._pipeline_graph import PipelineGraph, TaskImportMode
+from .visualization._options import NodeAttributeOptions
+from .visualization._show import show
+
+
+def main(argv: Sequence[str]) -> int:
+    """Run the development command-line interface.
+
+    Parameters
+    ----------
+    argv : `~collections.abc.Sequence` [ `str` ]
+        Commmand-line arguments, not including the program (typically
+        ``sys.argv[1:]``).
+
+    Notes
+    -----
+    This CLI is much more capable than pipetask's for very specific things, but
+    not as polished or user friendly.  It is intended primarily for use during
+    development of pipeline graph (especially visualization, where tests can't
+    really be used to judge the quality of the outputs, and a fast turnaround
+    is desirable).
+    """
+    parser = argparse.ArgumentParser(
```

Comment:
Please follow the approach for non-click command lines  https://developer.lsst.io/stack/argparse-script-topic-type.html and separate arparse creation from the `main` function.

Reply:
Will do (if we keep this; see main-PR-thread comment).